plugins/chat/app/controllers/chat/api/channel_threads_controller.rb

```diff
@@ -80,15 +80,16 @@ def create
           membership:,
         )
       end
-      on_failed_policy(:threading_enabled_for_channel) { raise Discourse::NotFound }
-      on_failed_policy(:can_view_channel) { raise Discourse::InvalidAccess }
-      on_failed_step(:create_thread) do |step|
-        render json: failed_json.merge(errors: [step.error]), status: 422
-      end
       on_failure { render(json: failed_json, status: 422) }
```

Comment:
I don't think it changes anything as such, but I wonder if `on_failure` be the last rule here since it acts as a catch all?